tests/components/water_heater/test_init.py

```diff
@@ -0,0 +1,12 @@
+"""Tests for Water heater."""
+from homeassistant.components import water_heater
+
+
+def test_deprecated_base_class(caplog):
+    """Test deprecated base class."""
+
+    class CustomWaterHeater(water_heater.WaterHeaterDevice):
+        pass
+
+    CustomWaterHeater()
+    assert "WaterHeaterDevice is deprecated, modify CustomWaterHeater" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5456697)